src/cmd_output.rs

```diff
@@ -39,11 +38,54 @@ impl CmdOutput for () {
     }
 }
 
+/// See the [`CmdOutput`] implementation for [`StdoutTrimmed`] below.
+#[derive(Debug, PartialEq, Clone)]
+pub struct StdoutTrimmed(pub String);
+
 /// Returns what the child process writes to `stdout`, interpreted as utf-8,
-/// collected into a string. This also suppresses output of the child's `stdout`
-/// to the parent's `stdout`. (Which would be the default when not using [`String`]
+/// collected into a string, trimmed of leading and trailing whitespace.
+/// This also suppresses output of the child's `stdout`
+/// to the parent's `stdout`. (Which would be the default when not using [`StdoutTrimmed`]
 /// as the return value.)
-impl CmdOutput for String {
+///
+/// It's recommended to pattern-match to get to the inner [`String`].
+/// This will make sure that the return type can be inferred.
+/// Here's an example:
+///
+/// ```
+/// use std::path::Path;
+/// use stir::*;
+///
+/// # #[cfg(unix)]
+/// # {
+/// let StdoutTrimmed(output) = cmd!("which ls");
+/// assert!(Path::new(&output).exists());
+/// # }
+/// ```
+impl CmdOutput for StdoutTrimmed {
+    fn prepare_config(config: &mut Config) {
+        <StdoutUntrimmed as CmdOutput>::prepare_config(config);
```

Comment:
Why doesn't this recurse?

Reply:
The implementation for `StdoutTrimmed` invokes the implementation for `StdoutUntrimmed`. Just to make it DRY. Do you it'd be clearer to just duplicate the implementation? Same question for `from_run_result`...

Reply:
Tweaked this a little after offline consultation.